modules/gapi/src/api/gcomputation.cpp

```diff
@@ -109,6 +111,20 @@ cv::GStreamingCompiled cv::GComputation::compileStreaming(GCompileArgs &&args)
     return comp.compileStreaming();
 }
 
+cv::GStreamingCompiled cv::GComputation::compileStreaming(const cv::detail::ExtractMetaCallback &callback,
+                                                                GCompileArgs                   &&args)
+{
+    // NB: Meta is unknown yet, need to obtain input/output shapes
+    // to properly unpack runtime argmunets that came from python
+    auto copy = args;
+    auto compiled = cv::gimpl::GCompiler(*this, {}, std::move(args)).compileStreaming();
+
+    auto metas = callback(compiled.priv().inInfo());
```

Comment:
So you compile the graph only to obtain this `inInfo`? Note the compilation may be a heavyweight process.

Is there an alternative way to obtain this `inInfo`?

Reply:
Fixed